bindata/network/network-metrics/003-prometheus.yaml

```diff
@@ -0,0 +1,83 @@
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    name: monitor-network
+  name: monitor-network
+  namespace: openshift-network-metrics
+spec:
+  endpoints:
+    - interval: 10s
+      port: metrics
+      metricRelabelings:
+        - sourceLabels: [pod]
+          targetLabel: __pod
+        - sourceLabels: [namespace]
+          targetLabel: __namespace
+        - sourceLabels: [source_pod]
+          targetLabel: pod
+        - sourceLabels: [source_namespace]
+          targetLabel: namespace
+      bearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token"
+      scheme: "https"
+      tlsConfig:
+        caFile: "/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt"
+        serverName: "network-metrics-service.openshift-network-metrics.svc"
+  selector:
+    matchLabels:
+      service: network-metrics-service
+  namespaceSelector:
+    matchNames:
+      - openshift-network-metrics
+---
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    prometheus.io/scrape: "true"
+    service.alpha.openshift.io/serving-cert-secret-name: metrics-daemon-secret
+  labels:
+    service: network-metrics-service
+  name: network-metrics-service
+  namespace: openshift-network-metrics
+spec:
+  selector:
+    app: network-metrics-daemon
+  clusterIP: None
+  ports:
+    - name: metrics
+      port: 8443
+      targetPort: https
+  type: ClusterIP
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: prometheus-k8s
+  namespace: openshift-network-metrics
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - services
+      - endpoints
+      - pods
```

Comment:
What is this role used for? Why does it need those permissions?

Reply:
It's required by the prometheus pod for service discovery (https://github.com/coreos/prometheus-operator/blob/master/Documentation/rbac.md#prometheus-rbac) 
In this case, we use a `ServiceMonitor` which means that prometheus monitors all the pods backing that service. I think prometheus needs those permission to perform such resolution.